### PR TITLE
Add timing instrumentation to the exporter

### PR DESCRIPTION
**Pull request by zmc**
This adds data for time spent while:
* locking nodes
* reimaging nodes
* running the bootstrap script
* running a job
* running each phase of each task in a job

### CODE REVIEW

**Comment by zmc** [comment]
@dmick this PR adds docs too :)

File: teuthology/dispatcher/__init__.py [hunk 1]
```diff
@@ -224,13 +225,16 @@ def match(proc):
...
-        tries=-1,
-        reimage=False,
-    )
+    machine_type = job_config["machine_type"]
+    count = len(job_config['roles'])
+    with exporter.NodeLockingTime.labels(machine_type, count).time():
+        lock_ops.block_and_lock_machines(
+            fake_ctx,
+            count,
+            job_config['machine_type'],
```
Comment (dmick) [inline teuthology/dispatcher/__init__.py:234]: since you now have machine_type, could replace this

Comment (zmc) [reply teuthology/dispatcher/__init__.py:234]: oh thank you, I definitely intended to

File: teuthology/kill.py [hunk 2]
```diff
@@ -87,7 +87,10 @@ def kill_job(run_name, job_id, archive_base=None, owner=None, skip_nuke=False):
         owner = job_info['owner']
     kill_processes(run_name, [job_info.get('pid')])
     if 'machine_type' in job_info:
-        teuthology.exporter.JobResults().record(job_info["machine_type"], job_info["status"])
+        teuthology.exporter.JobResults().record(
+            job_info["machine_type"],
+            job_info.get("status", "dead")
```
Comment (zmc) [inline teuthology/kill.py:92]: @kamoltat you were right about needing a change here

**Review by dmick — APPROVED** [review]
All looks good and right to me.

**Pull request merged**